app/js/datamodels.js

```diff
@@ -21,691 +21,699 @@
         };
 
-        // create derived metric
-        this.metric = MetricListService.getOrCreateDerivedMetric(this.name, derivedFunction);
-
-        this.updateScope(this.metric.data);
-    };
+        return MultipleCumulativeMetricTimeSeriesDataModel;
+    });
 
-    MultipleCumulativeMetricTimeSeriesDataModel.prototype.destroy = function () {
-        // remove subscribers and delete derived metric
-        MetricListService.destroyDerivedMetric(this.name);
-
-        // remove subscribers and delete base metrics
-        /*jslint unparam: true*/
-        $.each(this.metricDefinitions, function (key, definition) {
-            MetricListService.destroyMetric(definition);
-        });
-        /*jslint unparam: false*/
-
-        WidgetDataModel.prototype.destroy.call(this);
-    };
+    datamodels.factory('NetworkBytesMetricTimeSeriesDataModel', function (WidgetDataModel, MetricListService, VectorService) {
+        var NetworkBytesMetricTimeSeriesDataModel = function () {
+            return this;
+        };
 
-    return MultipleCumulativeMetricTimeSeriesDataModel;
-});
+        NetworkBytesMetricTimeSeriesDataModel.prototype = Object.create(WidgetDataModel.prototype);
 
-datamodels.factory('NetworkBytesMetricTimeSeriesDataModel', function (WidgetDataModel, MetricListService, VectorService) {
-    var NetworkBytesMetricTimeSeriesDataModel = function () {
-        return this;
-    };
+        NetworkBytesMetricTimeSeriesDataModel.prototype.init = function () {
+            WidgetDataModel.prototype.init.call(this);
 
-    NetworkBytesMetricTimeSeriesDataModel.prototype = Object.create(WidgetDataModel.prototype);
+            this.name = this.dataModelOptions ? this.dataModelOptions.name : 'metric_' + VectorService.getGuid();
 
-    NetworkBytesMetricTimeSeriesDataModel.prototype.init = function () {
-        WidgetDataModel.prototype.init.call(this);
+            // create create base metrics
+            var inMetric = MetricListService.getOrCreateCumulativeMetric('network.interface.in.bytes'),
+                outMetric = MetricListService.getOrCreateCumulativeMetric('network.interface.out.bytes'),
+                derivedFunction;
 
-        this.name = this.dataModelOptions ? this.dataModelOptions.name : 'metric_' + VectorService.getGuid();
+            // create derived function
+            derivedFunction = function () {
+                var returnValues = [],
+                    lastValue;
 
-        // create create base metrics
-        var inMetric = MetricListService.getOrCreateCumulativeMetric('network.interface.in.bytes'),
-            outMetric = MetricListService.getOrCreateCumulativeMetric('network.interface.out.bytes'),
-            derivedFunction;
+                /*jslint unparam: true*/
+                $.each(inMetric.data, function (index, instance) {
+                    if (instance.values.length > 0) {
+                        lastValue = instance.values[instance.values.length - 1];
```

Comment:
Good catch. This code was already there, though. This pull-request only addresses refactoring to improve the project structure (so I'm just moving code around, and cleaning up).